worktree_commit.go

```diff
@@ -0,0 +1,243 @@
+package git
+
+import (
+	"io"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/filemode"
+	"gopkg.in/src-d/go-git.v4/plumbing/format/index"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+	"gopkg.in/src-d/go-git.v4/storage"
+	"gopkg.in/src-d/go-git.v4/utils/ioutil"
+
+	"gopkg.in/src-d/go-billy.v2"
+)
+
+// Commit stores the current contents of the index in a new commit along with
+// a log message from the user describing the changes.
+func (w *Worktree) Commit(msg string, opts *CommitOptions) (plumbing.Hash, error) {
```

Comment:
I find the design of the commit operation very surprising:

- I would expect this method to receive what you want to commit (an index or whatever), along with the msg and the options.

- I would expect this to be a method of Repository or Storage or any other entity, except for worktree.

Maybe you should document the overall design of this somewhere.